src/hotspot/share/logging/logAsyncWriter.hpp

```diff
@@ -137,11 +137,9 @@ class AsyncLogWriter : public NonJavaThread {
   static AsyncLogWriter* _instance;
   // _lock(1) denotes a critional region.
   Semaphore _lock;
-  // _sem is a semaphore whose value denotes how many messages have been enqueued.
-  // It decreases in AsyncLogWriter::run()
-  Semaphore _sem;
   Semaphore _flush_sem;
-
+  os::PlatformMonitor _cv;
```

Comment:
No this isn't what you need. The _lock is the PlatformMonitor. You lock/unlock in AsyncLogLocker for the critical regions, and then use the wait/notify operations for the coordination.

Reply:
you mean I should use os::PlatformMonitor _lock as a combination of mutux and cv, right?